meson.build

```diff
@@ -0,0 +1,1629 @@
+    #'osdep/macos/log_helper.swift'
+    #'osdep/macos/libmpv_helper.swift'
+    #'osdep/main-fn-cocoa.c'
+    #'osdep/macos/mpv_helper.swift'
+    #'osdep/macos/swift_extensions.swift'
+    #'osdep/macos/swift_compat.swift'
+    #'osdep/macos/remote_command_center.swift' 'macos-media-player'
+    #'video/out/mac/common.swift'
+    #'video/out/mac/view.swift'
+    #'video/out/mac/window.swift'
+    #'video/out/mac/title_bar.swift'
+    #'video/out/cocoa_cb_common.swift', 'macos-cocoa-cb'
+    #'video/out/mac/gl_layer.swift', 'macos-cocoa-cb'
+
+
+    # TODO: dummy stuff
+
+    #'osdep/terminal-dummy.c'
+    #'input/ipc-dummy.c'
+    #'osdep/subprocess-dummy.c'
+
+
+    #'video/out/vulkan/context_android.c',  'vulkan && android',
+    #'video/out/vulkan/context_win.c',      'vulkan && win32-desktop',
+
+
+    #'video/d3d.c',                         'd3d-hwaccel',
+    #'video/filter/vf_d3d11vpp.c',          'd3d-hwaccel',
+
+
+    #'video/out/cocoa/events_view.m',       'cocoa',
+    #'video/out/cocoa/video_view.m',        'cocoa',
+    #'video/out/cocoa/window.m',            'cocoa',
+    #'video/out/cocoa_common.m',            'cocoa',
+    #'video/out/d3d11/context.c',           'd3d11',
+    #'video/out/d3d11/hwdec_d3d11va.c',     'd3d11 && d3d-hwaccel',
+    #'video/out/d3d11/hwdec_dxva2dxgi.c',   'd3d11 && d3d9-hwaccel',
+    #'video/out/d3d11/ra_d3d11.c',          'd3d11',
+    #'video/out/gpu/d3d11_helpers.c',       'd3d11 || egl-angle-win32',
+    #'video/out/opengl/context_cocoa.c',    'gl-cocoa',
+    #'video/out/opengl/context_rpi.c',      'rpi',
+    #'video/out/opengl/hwdec_d3d11egl.c',   'd3d-hwaccel && egl-angle',
+    #'video/out/opengl/hwdec_dxva2egl.c',   'd3d9-hwaccel && egl-angle',
+    #'video/out/opengl/hwdec_dxva2gldx.c',  'gl-dxinterop-d3d9',
+    #'video/out/opengl/hwdec_ios.m',        'ios-gl',
+    #'video/out/opengl/hwdec_osx.c',        'videotoolbox-gl',
+    #'video/out/opengl/hwdec_rpi.c',        'rpi-mmal',
+    #'video/out/vo_direct3d.c',             'direct3d',
+    #'video/out/vo_rpi.c',                  'rpi-mmal',
+
+
+    #'osdep/macosx_application.m',          'cocoa',
+    #'osdep/macosx_events.m',               'cocoa',
+    #'osdep/macosx_menubar.m',              'cocoa',
+    #'osdep/macosx_touchbar.m',             'macos-touchbar',
+    #'osdep/mpv.rc',                        'win32-executable',
+    #'osdep/path-macosx.m',                 'cocoa',
+
+if not get_option('cplayer').disabled()
+    datadir = get_option('datadir')
+    confdir = get_option('sysconfdir')
+
+    conf_files = ['etc/mpv.conf', 'etc/input.conf',
+                  'etc/mplayer-input.conf', 'etc/restore-old-bindings.conf']
+    install_data(conf_files, install_dir: join_paths(datadir, 'doc', 'mpv'))
+
+    zsh = find_program('zsh', required: false)
+    if zsh.found()
+        zsh_install_dir = join_paths(datadir, 'zsh', 'site-functions')
+        install_data('etc/_mpv.zsh', install_dir: zsh_install_dir, rename: '_mpv')
```

Comment:
You should really do this (and the bash completions) regardless of whether the shell is installed. bash will generally always be installed anyway, but for zsh you actually want to install the completions on Linux distro builds, so that it is available for end users who do have zsh, and are installing mpv via pacman or apt rather than cloning and building it themselves.